public/pages/workflow_detail/workspace/workspace.tsx

```diff
@@ -39,9 +46,17 @@ export function Workspace(props: WorkspaceProps) {
   const reactFlowWrapper = useRef(null);
   const { reactFlowInstance, setReactFlowInstance } = useContext(rfContext);
 
-  const [nodes, setNodes, onNodesChange] = useNodesState([]);
+  const [nodes, setNodes, onNodesChange] = useNodesState<IComponentData>([]);
   const [edges, setEdges, onEdgesChange] = useEdgesState([]);
 
+  // Listener for node additions or deletions to propagate to parent component
+  const nodesLength = useStore(
+    (state) => Array.from(state.nodeInternals.values()).length || 0
```

Comment:
would the nodes.length not be updated to use here?

Reply:
It's a custom hook that will be updated when the node internals change (node added/deleted/etc.). See docs and similar example: https://reactflow.dev/docs/api/hooks/use-store/

Reply:
And similar to the example, we listen for when the value changes on our useEffect on line 56-58